types/wlr_linux_dmabuf_v1.c

```diff
@@ -751,6 +740,104 @@ static void handle_renderer_destroy(struct wl_listener *listener, void *data) {
 	linux_dmabuf_v1_destroy(linux_dmabuf);
 }
 
+/* from the gamma example */
+static int create_anonymous_file(off_t size) {
+	char template[] = "/tmp/wlroots-shared-XXXXXX";
+	int fd = mkstemp(template);
+	if (fd < 0) {
+		return -1;
+	}
+
+	int ret;
+	do {
+		errno = 0;
+		ret = ftruncate(fd, size);
+	} while (errno == EINTR);
+	if (ret < 0) {
+		close(fd);
+		return -1;
+	}
+
+	unlink(template);
+	return fd;
+}
+
+static bool wlr_dmabuf_format_table_init(struct wlr_linux_dmabuf_v1 *linux_dmabuf,
+        struct wlr_renderer *renderer) {
+    int fd, idx = 0, len = 0;
+    struct wlr_dmabuf_format_table_entry *data;
+    size_t size;
+    uint16_t *index;
```

Comment:
Style: declare variables as needed, instead of doing it at the top of the function body